src/firefly/robot.py

```diff
@@ -0,0 +1,170 @@
+    def __init__(self):
+        self.setup_ui()
+
+    def setup_ui(self):
+        self.layout = QtWidgets.QHBoxLayout()
+
+        # First item, motor No.
+        # self.motor_label = QtWidgets.QLabel()
+        # self.motor_label.setText("1")
+        # self.layout.addWidget(self.motor_label)
+
+        # Second item, ComponentSelector
+        self.motor_box = ComponentSelector()
+        # self.stop_line_edit.setPlaceholderText("Stop…")
+        self.layout.addWidget(self.motor_box)
+
+        # Third item, start point
+        self.start_line_edit = QtWidgets.QLineEdit()
+        self.start_line_edit.setPlaceholderText("Position…")
+        self.layout.addWidget(self.start_line_edit)
+
+        # Forth item, stop point
+        # self.stop_line_edit = QtWidgets.QLineEdit()
+        # self.stop_line_edit.setPlaceholderText("Stop…")
+        # self.layout.addWidget(self.stop_line_edit)
+
+
+class RobotDisplay(display.FireflyDisplay):
+    """
+    A GUI for managing sample tranfer using a robot plan: robot_sample(robot, number/None, motor1, position1, motor2, position2, motor3, position...)
+    """
+
+    def customize_ui(self):
+        self.reset_default_regions()
+        # disable the line edits in spin box
+        # self.ui.robot_combo_box.lineEdit().setReadOnly(True)
+        # clear any exiting items in the combo box
+        # self.ui.robot_combo_box.clear()
+        # set the list of values for the combo box
+        # for rbt in ROBOT_NAMES:
+        #    self.ui.robot_combo_box.addItem(rbt)
+
+        # clear any exiting items in the combo box
+        self.ui.sample_combo_box.clear()
+        # set the list of values for the combo box
+        for sam in SAMPLE_NUMBERS:
+            self.ui.sample_combo_box.addItem(str(sam))
+        # self.ui.sample_combo_box.setCurrentIndex(0)
+
+        # disable the line edits in spin box
+        self.ui.num_motor_spin_box.lineEdit().setReadOnly(True)
+        self.ui.num_motor_spin_box.valueChanged.connect(self.update_regions)
+        # self.ui.num_motor_spin_box.editingFinished.connect(self.update_regions)
+
+        # self.ui.run_button.setEnabled(True) #for testing
+        self.ui.run_button.clicked.connect(self.queue_plan)
+
+    def reset_default_regions(self):
+        default_num_regions = 1
+        if not hasattr(self, "regions"):
+            self.regions = []
+            self.add_regions(default_num_regions)
+        self.ui.num_motor_spin_box.setValue(default_num_regions)
+        self.update_regions()
+
+    def add_regions(self, num=1):
+        for i in range(num):
+            region = LineScanRegion()
+            self.ui.regions_layout.addLayout(region.layout)
+            # Save it to the list
+            self.regions.append(region)
+
+    def remove_regions(self, num=1):
+        for i in range(num):
+            layout = self.regions[-1].layout
+            # iterate/wait, and delete all widgets in the layout in the end
+            while layout.count():
```

Comment:
Could this be ``while layout.count() > 0``? This current way should work okay, but checking explicitly for ``> 0`` makes it more clear what you're thinking.